LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionCN470.cs

```diff
@@ -119,10 +120,12 @@ public override bool TryGetJoinChannelIndex(double frequency, out int channelInd
         [Obsolete("#655 - This Rxpk based implementation will go away as soon as the complete LNS implementation is done.")]
         public override bool TryGetDownstreamChannelFrequency(Rxpk upstreamChannel, out double frequency, DeviceJoinInfo deviceJoinInfo)
         {
-            frequency = 0;
+            if (deviceJoinInfo is null) throw new ArgumentNullException(nameof(deviceJoinInfo));
```

Comment:
Wouldn't devicejoininfo be null in non china regions?

Reply:
Correct but this is in CN Region and here `deviceJoinInfo` is required, that's why I added the validation.